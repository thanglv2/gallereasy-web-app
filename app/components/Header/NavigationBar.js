import styled from 'styled-components';

const NavigationBar = styled.nav`
  top: 0;
  text-align: center;
  border-bottom: 1px solid ${props => props.theme.grey};
`;

const Menu = styled.ul`
  width: 100%;
  list-style-type: none;
  margin: 0;
  padding: 0;
  & > :first-child > a {
    color: ${props => props.theme.dimgrey};
    border-right: 2px solid ${props => props.theme.grey};
    display: block;
    padding: 0 25px;
  }

  @media(max-width: 600px) {
    & > :first-child > a {
      padding: 0 10px;
    }
  }
`;

export { NavigationBar, Menu };