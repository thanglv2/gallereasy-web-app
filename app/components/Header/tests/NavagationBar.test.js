import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import { NavigationBar, Menu } from '../NavigationBar';

describe('<NavigationBar />', () => {
  it('render without error', () => {
    const theme = {
      grey: 'grey'
    }
    const tree = renderer.create(<NavigationBar theme={theme} />).toJSON()
    expect(tree).toMatchSnapshot()
  });
});

describe('<Menu />', () => {
  it('render without error', () => {
    const theme = {
      dimgrey: 'dimgrey',
      grey: 'grey'
    }
    const tree = renderer.create(<Menu theme={theme} />).toJSON()
    expect(tree).toMatchSnapshot()
  });
});
