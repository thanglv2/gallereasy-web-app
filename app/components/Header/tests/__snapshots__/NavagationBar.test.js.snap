// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> render without error 1`] = `
.c0 {
  width: 100%;
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.c0 > :first-child > a {
  color: dimgrey;
  border-right: 2px solid grey;
  display: block;
  padding: 0 25px;
}

<ul
  className="c0"
/>
`;

exports[`<MenuItem /> render without error 1`] = `
.c0 {
  float: left;
  text-align: center;
  padding: 14px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: grey;
}

.c0.active a {
  color: black;
}

<li
  className="c0"
/>
`;

exports[`<NavigationBar /> render without error 1`] = `
.c0 {
  top: 0;
  text-align: center;
  border-bottom: 1px solid grey;
}

<nav
  className="c0"
/>
`;
